pkg/sidecar/util_posix.go

```diff
@@ -5,23 +5,54 @@ package sidecar
 
 import (
 	"context"
-	"errors"
 	"fmt"
+	"sync"
 
 	"github.com/spiffe/go-spiffe/v2/workloadapi"
 	"golang.org/x/sys/unix"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 // RunDaemon starts the main loop
 // Starts the workload API client to listen for new SVID updates
 // When a new SVID is received on the updateChan, the SVID certificates
 // are stored in disk and a restart signal is sent to the proxy's process
 func (s *Sidecar) RunDaemon(ctx context.Context) error {
-	err := workloadapi.WatchX509Context(ctx, &x509Watcher{sidecar: s}, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
-	if err != nil && !errors.Is(err, context.Canceled) {
-		return err
+	var wg sync.WaitGroup
+
+	if s.config.SvidFileName != "" && s.config.SvidKeyFileName != "" && s.config.SvidBundleFileName != "" {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := workloadapi.WatchX509Context(ctx, &x509Watcher{sidecar: s}, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
+			if err != nil && status.Code(err) != codes.Canceled {
+				s.config.Log.Fatalf("Error watching X.509 context: %w", err)
```

Comment:
Replace `Log.Fatalf` with `Log.Errorf` to prevent the application from exiting abruptly. Also use `%v` to properly format the error.

Use an error channel to collect errors from goroutines:

```
errChan := make(chan error, 1) 

if s.config.SvidFileName != "" && s.config.SvidKeyFileName != "" && s.config.SvidBundleFileName != "" {
	wg.Add(1)
	go func() {
		defer wg.Done()
		err := workloadapi.WatchX509Context(ctx, &x509Watcher{sidecar: s}, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
		if err != nil && status.Code(err) != codes.Canceled {
			s.config.Log.Errorf("Error watching X.509 context: %v", err)
			errChan <- err
		}
	}()
}
...

// Close the error channel when all goroutines are done.
go func() {
	wg.Wait()
	close(errChan)
}()

// Return the first error that occurs.
for err := range errChan {
	if err != nil {
		return err
	}
}
```

Apply to the whole method.